### DISCUSSION TITLE
How to get active translatable language

### DISCUSSION THREAD

**Discussion by ssglopes** [filamentphp/filament#6529] [Help]
With `Actions\LocaleSwitcher::make()` you get a select field with the language that are used for switching between languages for descriptions, titles etc. I looking for a way to retrieve the selected language so it can be used inside 
`Forms\Components\Select::make('tags.name'.{translatableLangHere()})` in the filament resource classes. So im not refering to `app()->getLocale()` which is the system language. Any example would be much appreciated. Thanks

**Comment by deveti** [comment]
What I did is, I created a settings page (spatie-settings), where I have Select with hardcoded values of "available" languages.
Then in AppServiceProvider boot method I added:

```
config([
    'filament-spatie-laravel-translatable-plugin.default_locales' => app(GeneralSettings::class)->site_languages
]);
```

You could make a new resource and use that to read languages from database for example.

**Comment by ArtMin96** [comment]
```php
$this->activeFormLocale
```
If I remember correctly, this is how you can get the active locale in the Filament.

**Comment by nabileisa** [comment]
Hi @ssglopes  , if you are still looking for  the answer , I have found it.
what you're looking for is a property called `activeLocale` and it exists on `$livewire`.
to access it you use a `Closure` and inject `$livewire`.
For example:
If you have a relationship called `type` and you want to display it's `name` in a `Forms\Components\Select` this is how I was able to make it  display the correct values in the locale chosen in the `LocaleSwithcher`
```php
// use the database field id
Forms\Components\Select::make('type_id')
// specify the relationship
->relationship('type', 'name')
// specify how to get the relationship translated values
->getOptionLabelFromRecordUsing(fn($record, $livewire) => $record->getTranslation('name', $livewire->activeLocale))
```
Hope this helps
And I hope this gets added to the docs to avoid future confusion.

**Reply by giuseppe-osenda** [reply]
Thank you very much! still not added in the doc!

**Comment by meharenderb** [comment]
If someone wants to conditionally validating required field based on selected language.
```php
Forms\Components\TextInput::make('slug')
    ->label('Slug')
    ->required(fn($livewire) =>$livewire->activeLocale == 'en')
``` 

In this code i required field when language is only 'en' otherwise required false.
